generator/templates/backend/ValidationUtils.cpp

```diff
@@ -0,0 +1,40 @@
+//* Copyright 2018 The NXT Authors
+//*
+//* Licensed under the Apache License, Version 2.0 (the "License");
+//* you may not use this file except in compliance with the License.
+//* You may obtain a copy of the License at
+//*
+//*     http://www.apache.org/licenses/LICENSE-2.0
+//*
+//* Unless required by applicable law or agreed to in writing, software
+//* distributed under the License is distributed on an "AS IS" BASIS,
+//* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//* See the License for the specific language governing permissions and
+//* limitations under the License.
+
+#include "backend/ValidationUtils_autogen.h"
+
+namespace backend {
+
+    {% for type in by_category["enum"] %}
+        bool IsValid{{type.name.CamelCase()}}(nxt::{{as_cppType(type.name)}} value) {
```

Comment:
Couldn't this be a "<" since our enums should be contiguous? Maybe with some extra asserts.

Reply:
I was hoping that the switch would be future proof for when we start to have sparse enums for extensions. Compilers do a good job of optimizing this pattern when it is dense, see https://godbolt.org/g/HjWVZH
WDYT?

Reply:
SGTM.